modeling-cmds/src/shared.rs

```diff
@@ -739,6 +739,55 @@ pub struct CameraSettings {
     pub ortho: bool,
 }
 
+#[allow(missing_docs)]
+#[repr(u8)]
+#[derive(Default, Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "ts-rs", derive(ts_rs::TS))]
+#[cfg_attr(feature = "ts-rs", ts(export_to = "ModelingCmd.ts"))]
+pub enum WorldCoordinateSystem {
+    #[default]
+    RightHandedUpZ,
+    RightHandedUpY,
+}
+
+#[allow(missing_docs)]
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[cfg_attr(feature = "ts-rs", derive(ts_rs::TS))]
+#[cfg_attr(feature = "ts-rs", ts(export_to = "ModelingCmd.ts"))]
+pub struct CameraViewState {
+    pub pivot_rotation: [f32; 4],
+    pub pivot_position: [f32; 3],
```

Comment:
I think this should use our dedicated Point type, because it's not just 3 arbitrary number, it represents a specific point in space, right?
```suggestion
    pub pivot_position: Point3d<f32>,
```

Reply:
I was avoiding it here since this type passes through an FFI boundary and `Point3d` isn't `repr(C)`. Might be erring on the side of caution unnecessarily though.